From cf63e2c447b95fec11a9660b0a925ba9a3fa1de6 Mon Sep 17 00:00:00 2001
From: Carlos Bentzen <cadubentzen@igalia.com>
Date: Tue, 10 Dec 2024 11:44:31 +0100
Subject: [PATCH] Combine upstream fixes for westerossink immediate-output

This patch combines the fixes for immediate-output in westerossink from
upstream in those commits:

commit 04c096cb3df0bc0797f5b6bc568e1d11b62dcb5b
Author: yramak601 <yashaswini.ramakrishna@rdkcentral.com>
Date:   Wed Feb 21 20:21:44 2024 +0000

    brcm: adjust decode rate for immediate output

    Set trickstate rate to 1.4 instead of 1.2 or 2.0 to allow for overflow recover and low latency decode without excessive frame dropping.

    Change-Id: Iebba3cc0d57a62bdf3e4971f3fc5a5685ba47fd9
    Signed-off-by: yramak601 <yashaswini.ramakrishna@rdkcentral.com>

commit ce422deacb5c18c0d2d36bb9dfb7aa149862f86e
Author: Manoj Bhatta <manoj_bhatta@comcast.com>
Date:   Tue Apr 9 14:55:42 2024 +0000

    brcm: adjust decode rate for immediate output

        Set trickstate rate to 2.0 to allow for overflow recover and low latency decode without excessive frame dropping.

    Change-Id: Ibff0a3093fc6b568eeded6ece562b4bf087c5778
---
 westeros-sink/brcm/westeros-sink-soc.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/westeros-sink/brcm/westeros-sink-soc.c b/westeros-sink/brcm/westeros-sink-soc.c
index 103e08e..af3aa2b 100644
--- a/westeros-sink/brcm/westeros-sink-soc.c
+++ b/westeros-sink/brcm/westeros-sink-soc.c
@@ -1418,7 +1418,7 @@ gboolean gst_westeros_sink_soc_paused_to_playing( GstWesterosSink *sink, gboolea
    sink->soc.videoPlaying = TRUE;
    UNLOCK(sink);
 
-   if( (sink->soc.stcChannel != NULL) && (sink->soc.codec != bvideo_codec_unknown) )
+   if( ((sink->soc.stcChannel != NULL) || (sink->soc.useImmediateOutput))&& (sink->soc.codec != bvideo_codec_unknown) )
    {
       #if (NEXUS_PLATFORM_VERSION_MAJOR>15)
       if ( sink->soc.codec == bvideo_codec_vp9 )
@@ -3395,12 +3395,12 @@ static void updateClientPlaySpeed( GstWesterosSink *sink, gfloat clientPlaySpeed
       return;
    }
 
-   /* use 1.2,  1.5 and up tends to drop frames unnecessarily, if they are only a bit late.
-      1.2 will get to the same low latency, but is smoother (and a bit slower) getting there than 2.0 */
+   /* use 2.0,  1.5 and up tends to drop frames unnecessarily, if they are only a bit late.
+      2.0 will get to the same low latency, but is smoother  */
    if ( sink->soc.useImmediateOutput )
    {
        NEXUS_SimpleVideoDecoder_GetTrickState(sink->soc.videoDecoder, &trickState);
-       trickState.rate= playing ? NEXUS_NORMAL_DECODE_RATE * 1.2 : 0.0;
+       trickState.rate= playing ? NEXUS_NORMAL_DECODE_RATE * 2.0 : 0.0;
        trickState.tsmEnabled= NEXUS_TsmMode_eDisabled;
        GST_LOG("updateClientPlaySpeed: useImmediateOutput: SetTrickState rate %d TsmMode disabled", trickState.rate);
        NEXUS_SimpleVideoDecoder_SetTrickState(sink->soc.videoDecoder, &trickState);
-- 
2.47.1

